lib/WTSI/NPG/HTS/PacBio/Sequel/RunDelete.pm

```diff
@@ -0,0 +1,97 @@
+package WTSI::NPG::HTS::PacBio::Sequel::RunDelete;
+
+use namespace::autoclean;
+use Moose;
+use MooseX::StrictConstructor;
+use Readonly;
+use English qw[-no_match_vars];
+
+Readonly::Scalar my $PROD_DIR_COUNT => 5;
+Readonly::Scalar my $PACBIO_FOLDER_NAME => q[pacbio];
+
+with qw[
+         WTSI::DNAP::Utilities::Loggable
+       ];
+
+our $VERSION = '';
+
+has 'runfolder_path' =>
+  (isa           => 'Str',
+   is            => 'ro',
+   required      => 1,
+   documentation => 'PacBio runfolder path');
+
+
+=head2 delete_run
+
+  Arg [1]    : Check format option for production runs
+  Example    : $deleter->delete_run({runfolder_path => $path});
+  Description: Return true if successfully deleted
+  Returntype : Bool
+
+=cut
+
+sub delete_run {
+    my ($self, $checkformat) = @_;
+
+    defined $self->runfolder_path or
+        $self->logconfess('A defined runfolder is required');
+
+    ## check path conforms to production path format if required
+    if(defined $checkformat && $checkformat == 1) {
+        my @fields = split /\//mxs, $self->runfolder_path;
+         if( @fields != $PROD_DIR_COUNT ||
+            $fields[3] ne $PACBIO_FOLDER_NAME ) {
+            $self->logcroak('Folder failed format checks '. $self->runfolder_path);
+        }
+    }
+
+    ## touch first to check runfolder permissions
+    my $touch = q{touch }. $self->runfolder_path;
+    if (system($touch) != 0) {
+        my $e = $CHILD_ERROR || q[];
+        $self->logcroak(qq[Aborting deletion - error $e running "$touch"]);
+    }
+
+    ## then remove
+    my $remove = q{rm -rf }. $self->runfolder_path;
```

Comment:
Consider `File::Path::remove_tree`? `File::Path` is already a dependency and does a number of sanity checks while deleting.

Reply:
There are cases where remove_tree is very useful but not convinced of any major advantage here currently. Happy to change if you feel strongly ?

Reply:
Not strongly. It can do better than `rm -rf` if there are odd permissions in there, but we don't get that in practice because we already control the permissions ...